lang/semgrep-grammars/src/semgrep-dockerfile/test/corpus/semgrep.txt

```diff
@@ -65,3 +65,380 @@ RUN ["a", $X, "b"]
       (double_quoted_string)
       (semgrep_metavariable)
       (double_quoted_string))))
+
+=============================================
+Param metavariable
+=============================================
+
+FROM --platform=$PLATFORM debian
+ADD --chown=$USER_GROUP src dst
+
+---
+
+(source_file
+  (from_instruction
+    (param)
+    (image_spec
+      (image_name)))
+  (add_instruction
+    (param)
+    (path)
+    (path)))
+
+=============================================
+Image metavariable
+=============================================
+
+FROM $NAME:$TAG@$DIGEST
+FROM ${NAME}:${TAG}@${DIGEST}
+
+---
+
+(source_file
+  (from_instruction
+    (image_spec
+      (image_name
+        (expansion
```

Comment:
I thought $NAME here would be not considered a variable expansion but a metavariable.
Is it because there's a : after?

Reply:
`$NAME` and `${NAME}` are docker argument expansions. We'll inspect the token when converting the CST into a semgrep pattern to determine if it's a metavariable.

Reply:
This could be a useful comment to put in the test file.

Reply:
yes. I don't know how to put comments in tree-sitter test files. I'd rather avoid using the target language's comment syntax. I asked here: https://github.com/tree-sitter/tree-sitter/discussions/1586